### ISSUE TITLE
Support for integer values in valueQuantity

### ISSUE THREAD

**Issue by joarb** [FirelyTeam/spark#376]
In the R4 specification the Quantity (https://www.hl7.org/fhir/datatypes.html#Quantity) defines the attribute `value` of type decimal (https://www.hl7.org/fhir/datatypes.html#decimal).

According to the regex defining the decimal type, `-?(0|[1-9][0-9]*)(\.[0-9]+)?([eE][+-]?[0-9]+)?`, it should be possible to express integer values in the Quantity data type. Is it possible to output quantities as integers?

**Comment by kennethmyhra** [comment]
You could treat it as an integer if that's what you want?

Spark follows the specification and will therefore treat it as a decimal value.

**Comment by joarb** [comment]
Thanks! But is it possible for the json output to be formatted as an integer?

**Comment by kennethmyhra** [comment]
I'm a bit guessing now on what you want to achieve. I did some tests to see if I could store an `Observation` resource where I set `valueQuantity` with a value of `176` and then try to retrieve it later and verify that the value is still `176` and not i.e. `176.0`

```
{
    "resourceType": "Observation",
    "id": "r2-integer",
    "status": "final",
    "code": {
        "coding": [
            {
                "system": "http://loinc.org",
                "code": "718-7",
                "display": "Hemoglobin [Mass/volume] in Blood"
            }
        ],
        "text": "Haemoglobin"
    },
    "valueQuantity": {
        "value": 176,
        "unit": "g/L",
        "system": "http://unitsofmeasure.org",
        "code": "g/L"
    }
}
```

When retrieving that resource later valueQuantity now has a decimal point and is returned as `176.0` which is probably not what we want

```
"valueQuantity": {
    "value": 176.0,
    "unit": "g/L",
    "system": "http://unitsofmeasure.org",
    "code": "g/L"
}
```

I have not done any further investigation or testing, but this is most probably happening in the MongoDB BSON serializers. It could probably be tweaked by modifying the registered BSON serializer for the decimal type or by implementing your own `BsonDecimalSerializer` and register that using the method `BsonSerializer.RegisterSerializer` on start up.

A third alternative is implementing an `IBsonSerializationProvider` like I have done in Spark.Mongo.Tests: https://github.com/FirelyTeam/spark/blob/stu3/master/src/Spark.Mongo.Tests/Search/BsonSerializationProvider.cs and then register that using `BsonSerializer.RegisterSerializationProvider`. As with the first or second alternative you would need to either tweak the existing BSON decimal serializer or implement your own BsonDecimalSerializer and return that from `IBsonSerializationProvider.GetSerializer`
